### PR TITLE
Workaround the UTF-8 encoding restriction of google.rpc.Status protobuf mssage

### PR DESCRIPTION
**Pull request by veblush**
Protobuf message requires the text field to be a valid UTF-8 string and gRPC serializes child `absl::Status` to store it at the parent Status. This is mostly okay but upb will return an error for the parsing error which happens to have invalid UTF-8 character. This is caught by a fuzzer test. Solution is using payload instead of message because payload is bytes which doesn't care its encoding.

In addition to this, Parsing message is updated not to have non UTF-8 characters because it's still valuable to keep UTF-8 messages

### CODE REVIEW

File: test/core/gprpp/status_helper_test.cc
Comment (markdroth) [inline test/core/gprpp/status_helper_test.cc:107]: What if the text has a lot of characters that are not UTF-8?  Couldn't this basically render the message unreadable?

**Review by markdroth — COMMENTED** [review]
I'd like to get input from @ctiller on this.  I'm not sure if there are any implications for status conversion.

File: test/core/gprpp/status_helper_test.cc
Comment (veblush) [reply test/core/gprpp/status_helper_test.cc:107]: Yes it will be replaced with space, maybe not readable but I don't think the original message with code (>= 128) is readable either without a proper encoding.

Comment (veblush) [reply test/core/gprpp/status_helper_test.cc:107]: One thing I can do alternatively is to serialize the message field in a payload. In this way, it doesn't need to be sanitized because its type is bytes.

Comment (veblush) [reply test/core/gprpp/status_helper_test.cc]: I changed this PR to store the message in the payload to workaround this limitation. PTAL.

File: src/core/lib/gprpp/status_helper.cc
Comment (markdroth) [inline src/core/lib/gprpp/status_helper.cc:354]: Suggest changing this to something like `"grpc.utf8_message"`, to make its purpose clearer.

**Comment by markdroth** [comment]
Also, please do not force-push in the middle of a review.  That makes it much harder for the reviewer to see what's changed since their last review.  Thanks!

File: src/core/lib/gprpp/status_helper.cc
Comment (veblush) [reply src/core/lib/gprpp/status_helper.cc]: Done! But i removed utf8_ because it's not utf-8

File: src/core/ext/transport/chttp2/transport/parsing.cc
Comment (veblush) [reply src/core/ext/transport/chttp2/transport/parsing.cc:99]: It's still valuable to keep this change because although we don't need to technically but having utf-8 compatible strings is worth it.

**Comment by veblush** [comment]
> Also, please do not force-push in the middle of a review. That makes it much harder for the reviewer to see what's changed since their last review. Thanks!

Oh sorry about that. (I just did it again.)

File: src/core/ext/transport/chttp2/transport/parsing.cc
Comment (markdroth) [reply src/core/ext/transport/chttp2/transport/parsing.cc:99]: The way this is being done means that the message will actually be corrupted if it's composed of non-UTF-8 characters.  If you just allow the non-UTF-8 characters here, then the change you're making in `StatusToProto()` will properly propagate the error message.  But with the change here, the error message will be corrupted, and people may not be able to read it.

**Comment by veblush** [comment]
@ctiller Yes, that's a good idea. Once we finish the grpc-error migraiton, we can add that limitation.

**Comment by markdroth** [comment]
I already replied to your latest changes above.  I'd prefer to not modify the string in the chttp2 code, because that can basically corrupt the error message such that it's unusable for users.

File: src/core/ext/transport/chttp2/transport/parsing.cc
Comment (veblush) [reply src/core/ext/transport/chttp2/transport/parsing.cc:99]: I don't understand why you think this message is corrupted. It shows both the code of character and character itself only when the character falls in the valid utf-8 range. like `A (65)`. For the case of invalid utf-8 code, it used to print like `Æ (198)' but it changes to ` (198)` by this change. It sill carries the information about the code 198 but it just doesn't print it as a character because it's not a utf-8 code.

Comment (markdroth) [reply src/core/ext/transport/chttp2/transport/parsing.cc:99]: Oh, sorry, I was misreading this.  You're right, there is enough info here.

**Review by markdroth — APPROVED** [review]
This looks fine to me, but please get Craig's approval as well before merging.

**Comment by veblush** [comment]
@ctiller It's ready for your review. PTAL.

**Review by ctiller — APPROVED** [review]

**Pull request merged**
